qiskit/pulse/builder.py

```diff
@@ -0,0 +1,1986 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+r"""Use the pulse builder to write pulse programs with an imperative,
+assembly-like syntax.
+
+.. warning::
+    The pulse builder interface is still in active development and may have
+    breaking API changes without deprecation warnings in future releases until
+    otherwise indicated.
+
+To begin pulse programming we must first initialize our program builder
+context with :func:`build`, after which we can begin adding program
+statements. For example, below we write a simple program that :func:`play`\s
+a pulse:
+
+.. jupyter-execute::
+
+    from qiskit import execute
+    from qiskit import pulse
+
+    d0 = pulse.DriveChannel(0)
+
+    with pulse.build() as pulse_prog:
+        pulse.play(pulse.Constant(100, 1.0), d0)
+
+    pulse_prog.draw()
+
+The builder initializes a :class:`pulse.Schedule`, ``pulse_prog``
+and then begins to construct the program within the context. The output pulse
+schedule will survive after the context is exited and can be executed like a
+normal Qiskit program using ``qiskit.execute(pulse_prog, backend)``.
```

Comment:
This could be distracting for new users, or physicists without a software background

Reply:
I think this information is necessary at the current builder stage. This **should** be addressed with additional documentation at a later stage.